resources/plugins/hello/plugin.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+import json
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+import click
+
+from warnet.constants import PLUGIN_ANNEX, AnnexMember, HookValue, WarnetContent
+from warnet.process import run_command
+
+# Tt is common for Warnet objects to have a "mission" label to help query for them in the cluster.
+MISSION = "hello"
+PRIMARY_CONTAINER = MISSION
+
+PLUGIN_DIR_TAG = "plugin_dir"
+
+
+class PluginError(Exception):
+    pass
+
+
+log = logging.getLogger(MISSION)
+if not log.hasHandlers():
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+    console_handler.setFormatter(formatter)
+    log.addHandler(console_handler)
+log.setLevel(logging.DEBUG)
+log.propagate = True
+
+
+# Plugins look like this in the `network.yaml` file:
+#
+# plugins:
+#   hello:
+#     podName: "a-pod-name"
+#     helloTo: "World!"
+#
+# "podName" and "helloTo" are essentially dictionary keys, and it helps to keep those keys in an
+# enum in order to prevent typos.
+class PluginContent(Enum):
+    POD_NAME = "podName"
+    HELLO_TO = "helloTo"
+
+
+# Warnet uses a python package called "click" to manage terminal interactions with the user.
+# To use click, we must declare a click "group" by decorating a function named after the plugin.
+# While optional, using click makes it easy for users to interact with your plugin.
+@click.group()
+@click.pass_context
+def hello(ctx):
+    """Commands for the Hello plugin"""
+    ctx.ensure_object(dict)
+    plugin_dir = Path(__file__).resolve().parent
+    ctx.obj[PLUGIN_DIR_TAG] = Path(plugin_dir)
+
+
+# Each Warnet plugin must have an entrypoint function which takes two JSON objects: plugin_content
+# and warnet_content. We have seen the PluginContent enum above. Warnet also has a WarnetContent
```

Comment:
does the user need to provide warnet content? seems like the biggest advantage of the plugin architecture is direct access to the cluster

Reply:
We could make the warnet content optional. Would that solve this?